src/Common/tests/Tests/System/StringTests.cs

```diff
@@ -3837,24 +3839,46 @@ public static void Join_ObjectArray_Null_ThrowsArgumentNullException()
         [InlineData("", 'H', 0, 0, -1)]
         public static void LastIndexOf_SingleLetter(string s, char value, int startIndex, int count, int expected)
         {
+            int startPos = count == 0 ? 0 : startIndex - count + 1;
+            ReadOnlySpan<char> span = s.AsSpan(startPos, count);
+            int expectedFromSpan = expected - startPos;
+            ReadOnlySpan<char> valueSpan = value.ToString().AsSpan();
+
             if (count == s.Length)
             {
                 if (startIndex == s.Length - 1)
                 {
                     Assert.Equal(expected, s.LastIndexOf(value));
                     Assert.Equal(expected, s.LastIndexOf(value.ToString()));
                 }
-                Assert.Equal(expected, s.LastIndexOf(value, startIndex));
+                Assert.Equal(expectedFromSpan, span.LastIndexOf(value));
                 Assert.Equal(expected, s.LastIndexOf(value.ToString(), startIndex));
             }
             Assert.Equal(expected, s.LastIndexOf(value, startIndex, count));
+            Assert.Equal(expectedFromSpan, span.LastIndexOf(value));
             Assert.Equal(expected, s.LastIndexOf(value.ToString(), startIndex, count));
+            Assert.Equal(expectedFromSpan, span.LastIndexOf(valueSpan));
 
             Assert.Equal(expected, s.LastIndexOf(value.ToString(), startIndex, count, StringComparison.CurrentCulture));
+#if !FEATURE_PORTABLE_SPAN
```

Comment:
I'd prefer we not add such a define, and instead, if we can't add the overload for the portable version, add it as an extension method in the tests... worst case that extension method falls back to using strings, but at least the tests don't need to be littered with all of these ifdefs, which are difficult to maintain.

Reply:
Right, I considered that, but I expect this only to be temporary and be removed when the netstandard api set is unfrozen.

Reply:
> when the netstandard api set is unfrozen

When is that going to happen?  Doesn't seem like it's imminent.

Reply:
> When is that going to happen?

@ericstj probably knows.

> we'd want to do at that point is delete the extension method, rather than go through and delete all of these ifdefs.

Right, the extension method solution is more elegant. I will change this.

Reply:
> when the netstandard api set is unfrozen

The netstandard2.0 targeting version of the public surface is fronzen till *forever*. We may just want to stop building it and get it via package harvesting instead to avoid this complexity.

And once these is netstandard3.0, this API will be part of the platform and we are not going to need to build a netstandard targeting version either.

Reply:
right, I was also thinking about harvesting to get rid of this. Thanks, I didn't know that we will stop building a netstandard targeting version but aligning with our goals for netstandard3.0 this makes sense.